### PR TITLE
965 the old submission_competing_interests has been removed, this is controlled by submission configuration

### LINKED ISSUE #965: Duplicate Setting
Setting `submission_competing_interests` is a duplicate now handled by the Submission Fields Config and should be deleted.

### PR DESCRIPTION
**Pull request by ajrbyers**
- Note: As the interface required both settings to be enabled in order to work we can safely just replace `submission_competing_interests` with the configuration equivalent. Anyone actually using this setting wont see any difference.
- Closes #965

### CODE REVIEW

File: src/journal/migrations/0060_alter_journal_code.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+    dependencies = [
+        ('journal', '0059_alter_prepublicationchecklistitem_completed_by'),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='journal',
+            name='code',
+            field=models.CharField(help_text='Short acronym for the journal. Used as part of the journal URLin path mode and to uniquely identify the journal', max_length=40, unique=True),
```
Comment (mauromsl) [inline src/journal/migrations/0060_alter_journal_code.py:16]: This leaked from another branch

File: src/journal/models.py [hunk 2]
```diff
@@ -75,7 +75,7 @@ def issue_large_image_path(instance, filename):
 
 
 class Journal(AbstractSiteModel):
-    code = models.CharField(max_length=24, unique=True, help_text=gettext(
+    code = models.CharField(max_length=40, unique=True, help_text=gettext(
```
Comment (mauromsl) [inline src/journal/models.py:78]: This leaked from another branch

**Review by mauromsl — COMMENTED** [review]
Looks good, but there are a couple of unrelated changes leaked from another branch

File: src/submission/forms.py [hunk 3]
```diff
@@ -195,6 +196,12 @@ def __init__(self, *args, **kwargs):
                         except models.FieldAnswer.DoesNotExist:
                             pass
 
+                # if the editor is viewing the page, don't set additional
+                # fields to be required.
+                if editor_view:
+                    self.fields[element.name].required = False
+
+
```
Comment (joemull) [inline src/submission/forms.py:204]: Does this belong on this branch? And does it need to be indented further so that it is part of the for loop?

**Review by joemull — CHANGES_REQUESTED** [review]
There are still leaked changes, unless I'm mistaken.

**Comment by ajrbyers** [comment]
Commits in to revert weirdness.

**Review by joemull — CHANGES_REQUESTED** [review]

**Review by joemull — APPROVED** [review]

**Comment by ajrbyers** [comment]
Closed in favour of https://github.com/BirkbeckCTP/janeway/pull/3569

**Pull request closed without merging**